state/upgrades.go

```diff
@@ -815,3 +815,64 @@ func addStorageInstanceConstraints(st *State) error {
 	}
 	return nil
 }
+
+var splitBatchRemoveSize = 1000
+
+// SplitLogCollections moves log entries from the old single log collection
+// to the log collection per environment.
+func SplitLogCollections(st *State) error {
+	session := st.MongoSession()
+	db := session.DB(logsDB)
+	oldLogs := db.C("logs")
```

Comment:
I'll add a check that the collection exists first - I have a feeling that the remove all might raise an error if the collection doesn't exist. (I'll add a test.)